From a7f20943a547f3e3c28b9c52dd2996135c330ecd Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Wed, 8 Apr 2020 22:51:30 +0300
Subject: [PATCH 1/1] Attempt to symlink .config/Code to .config/Code - OSS if
 Code does not exist. This ensures some extensions that specifically check for
 Code will function. Additionally don't bother doing the platform checks.
 We're on Linux.

---
 src/paths.js | 23 ++++++++++++++++-------
 1 file changed, 16 insertions(+), 7 deletions(-)

diff --git a/src/paths.js b/src/paths.js
index a6d3c05228..24ca3e66e4 100644
--- a/src/paths.js
+++ b/src/paths.js
@@ -8,6 +8,7 @@
 
 // @ts-expect-error
 const pkg = require('../package.json');
+const fs = require('fs');
 const path = require('path');
 const os = require('os');
 
@@ -16,12 +17,7 @@ const os = require('os');
  * @returns {string}
  */
 function getAppDataPath(platform) {
-	switch (platform) {
-		case 'win32': return process.env['VSCODE_APPDATA'] || process.env['APPDATA'] || path.join(process.env['USERPROFILE'], 'AppData', 'Roaming');
-		case 'darwin': return process.env['VSCODE_APPDATA'] || path.join(os.homedir(), 'Library', 'Application Support');
-		case 'linux': return process.env['VSCODE_APPDATA'] || process.env['XDG_CONFIG_HOME'] || path.join(os.homedir(), '.config');
-		default: throw new Error('Platform not supported');
-	}
+	return process.env['VSCODE_APPDATA'] || process.env['XDG_CONFIG_HOME'] || path.join(os.homedir(), '.config');
 }
 
 /**
@@ -29,7 +25,20 @@ function getAppDataPath(platform) {
  * @returns {string}
  */
 function getDefaultUserDataPath(platform) {
-	return path.join(getAppDataPath(platform), pkg.name);
+	var appDataPath = getAppDataPath(platform);
+	var codePath = path.join(appDataPath, pkg.name);
+
+	if (pkg.name == "Code - OSS") { // If we're using OSS equivalent
+		var codeNonOSSPath = path.join(appDataPath, "Code"); // Get the path for the non-Code equivalent
+
+		fs.access(codeNonOSSPath, fs.constants.F_OK, (error) => { // Access the folder with only the file exists constant
+			if (error) { // If the file does not exist
+				fs.symlinkSync(codePath, codeNonOSSPath); // Symlink the two
+			}
+		});
+	}
+
+	return codePath;
 }
 
 exports.getAppDataPath = getAppDataPath;
-- 
2.26.0

